packages/sdk-core/src/Framework.ts

```diff
@@ -249,18 +264,33 @@ export default class Framework {
     };
 
     /**
-     * Load a `SuperToken` class from the `Framework`.
+     * Loads `SuperTokenWithUnderlying`, `SuperTokenWithoutUnderlying` or `NativeAssetSuperToken` class from the `Framework`.
      * @param tokenAddressOrSymbol the `SuperToken` address or symbol (if symbol, it must be on the resolver)
      * @returns `SuperToken` class
      */
-    loadSuperToken = async (
+    loadSuperToken = async <T extends SuperToken>(
```

Comment:
I personally think `<T extends SuperToken>` is a bit confusing.

I don't see how `await framework.loadSuperToken<SuperTokenWithUnderlying>("DAIx")` is much better than `await framework.loadSuperToken("DAIx") as SuperTokenWithUnderlying`.

If we were to make it a generic then I would expect more benefit or type-checking from it. Like it should break when I put `NativeAssetSuperToken` but it's not actually one. (It might not be possible to add any more type-checking though...) Right now it's just another way to cast the result.

But if we keep it like this then I would at least set the default type to the base class like so: `T extends SuperToken = SuperToken` because it shouldn't always be necessary to know beforehand whether it's a super token with underlying or not.

Reply:
okay-this is fair